core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -78,6 +78,9 @@ private TableProperties() {
   public static final String SPLIT_OPEN_FILE_COST = "read.split.open-file-cost";
   public static final long SPLIT_OPEN_FILE_COST_DEFAULT = 4 * 1024 * 1024; // 4MB
 
+  public static final String SPLIT_BY_PARTITION = "read.split.by-partition";
```

Comment:
It may not always be desired even if the table is partitioned. For instance, if the partitions of a table only contain one 4MB file each, sometimes it may be more beneficial to combine them into larger splits, especially when you know that the table is not used in a join, or that it is not joined against another table that has the same partitioning (and thus cannot leverage bucket join).

Reply:
I'm also skeptical that this is useful at the table level. It seems like this would always be something to configure during a scan. You only know that the table is or is not used in a join in the query engine. And we combine by default for the 4MB file case.

Reply:
I see what you mean now. Yes agree that it's not very useful on the table level, but we still want it as one of the Spark read options, is that correct?

Reply:
The problem with read options is that it requires modifying code to change the value. I am also not sure whether having a table property for this is going to help us much. Having this at table level will probably also affect other engines that may not necessarily benefit from bucketed joins.

I am not sure how we can detect whether a table is used in a join or not, though. I don't think Spark propagates that info to sources. Are there any ideas for that?

Overall, I am fine either way. I think we will need a read option, though. It will give us a way to force a particular value. We may want to default it for bucketed tables to true by default.

Reply:
Yea, I think it's not a good idea to have this as a table property, will update.

I'll keep it as a Spark read option if ppl think otherwise (cc @rdblue @prodeezy). 

> We may want to default it for bucketed tables in Spark to true unless set by the user in options.

I assume "bucketed" = "partitioned" :) since the storage-partitioned join on Spark side could apply to other partition specs than bucketing. If a table is unpartitioned then I think the behavior is the same no matter the flag is on or off. Therefore, I think we just need to set the default value to true for Spark?

Reply:
Let me resolve this thread as I think we've reached consensus on not having it as a table property. We will add a SQL config instead.